.github/workflows/ci_tests.yml

```diff
@@ -241,66 +205,112 @@ jobs:
             BUILD_AND_TEST_ARGS: "--build-exe-only --disable-unit-tests --disable-schema-deployment"
             HOST_CONFIG: host-configs/TOTAL/pecan-GPU.cmake
             GCP_BUCKET: geosx/Pecan-GPU
+    uses: ./.github/workflows/build_and_test.yml
+    with:
+      GEOSX_TPL_TAG: ${{ needs.check_pull_request_is_not_a_draft.outputs.GEOSX_TPL_TAG }}
+      DOCKER_REPOSITORY: ${{ matrix.DOCKER_REPOSITORY }}
+      CMAKE_BUILD_TYPE: ${{ matrix.CMAKE_BUILD_TYPE }}
+      BUILD_AND_TEST_ARGS: ${{ matrix.BUILD_AND_TEST_ARGS }}
+      HOST_CONFIG: ${{ matrix.HOST_CONFIG }}
+      ENABLE_HYPRE: ${{ matrix.ENABLE_HYPRE }}
+      ENABLE_HYPRE_DEVICE: ${{ matrix.ENABLE_HYPRE_DEVICE }}
+      ENABLE_TRILINOS: ${{ matrix.ENABLE_TRILINOS }}
+      GCP_BUCKET: ${{ matrix.GCP_BUCKET }}
+      COMMIT: ${{ github.event.pull_request.head.sha }}
+      OS: ${{ matrix.OS }}
+    secrets: inherit
+
+  # Note: cuda_builds_merge_not_ready and cuda_builds_merge_ready jobs
+  # should have identical matrix.include entries.
+  # If "flag: ready to be merged" PR label found, cuda jobs run
+  # immediately along side linux jobs.
+  cuda_builds_merge_ready:
+    name: ${{ matrix.name }}
+    needs: [check_pull_request_is_not_a_draft]
+    if: contains( github.event.pull_request.labels.*.name, format('flag{0} ready to be merged', ':'))
+    strategy:
 
-    steps:
-    - name: Checkout Repository
-      uses: actions/checkout@v3
-      with:
-        submodules: true
-        lfs: false
-
-    - id: 'auth'
-      if: matrix.GCP_BUCKET
-      uses: 'google-github-actions/auth@v1'
-      with:
-        credentials_json: '${{ secrets.GOOGLE_CLOUD_GCP }}'
-    - name: 'Set up Cloud SDK'
-      if: matrix.GCP_BUCKET
-      uses: 'google-github-actions/setup-gcloud@v1'
-      with:
-        version: '>= 363.0.0'
-
-    - name: Print environment
-      run: printenv
+      # In-progress jobs will not be cancelled if there is a failure
+      fail-fast : false
+      matrix:
```

Comment:
Is it possible to make an additional workflow "template" (like `build_and_test.yml`) which would call `build_and_test.yml` backend, with one unique source of parameters? (a template of template...).
Then we could call this template if the `if` branch and in the `if not` as well without any duplication.
If not, then I guess that this is what it is.

Reply:
OK, let's keep it like it is. `yaml` is the death of computer science.